product_contract/views/sale_order.xml

```diff
@@ -0,0 +1,124 @@
+                    domain="[('contract_id','=',contract_id)]"
+                />
+            </xpath>
+            <xpath
+                expr="//field[@name='order_line']/form//field[@name='tax_id']/parent::group"
+                position="after"
+            >
+                <field name="is_contract" attrs="{'invisible': []}" />
+                <separator
+                    colspan="4"
+                    string="Recurrence Invoicing"
+                    attrs="{'invisible': [('is_contract', '=', False)]}"
+                />
+                <group attrs="{'invisible': [('is_contract', '=', False)]}">
+                    <field name="recurring_rule_type" />
+                </group>
+                <group attrs="{'invisible': [('is_contract', '=', False)]}">
+                    <field name="recurring_invoicing_type" />
+                </group>
+                <group attrs="{'invisible': [('is_contract', '=', False)]}">
+                    <field
+                        name="date_start"
+                        attrs="{'required': [('is_contract', '=', True)]}"
+                    />
+                </group>
+                <group attrs="{'invisible': [('is_contract', '=', False)]}">
+                    <field
+                        name="date_end"
+                        attrs="{'required': [('is_contract', '=', True)]}"
+                    />
+                </group>
+                <group attrs="{'invisible': [('is_contract', '=', False)]}">
+                    <field name="is_auto_renew" />
+                </group>
+                <group attrs="{'invisible': [('is_auto_renew', '=', False)]}">
+                    <label for="auto_renew_interval" />
+                    <div>
+                        <field
+                            name="auto_renew_interval"
+                            class="oe_inline"
+                            nolabel="1"
+                            attrs="{'required':[('is_auto_renew', '=', True)]}"
+                        />
+                        <field
+                            name="auto_renew_rule_type"
+                            class="oe_inline"
+                            nolabel="1"
+                            attrs="{'required':[('is_auto_renew', '=', True)]}"
+                        />
+                    </div>
+                </group>
+            </xpath>
+            <xpath
+                expr="//field[@name='order_line']/tree//field[@name='price_total']"
+                position="after"
+            >
+                <field
+                    name="date_start"
+                    attrs="{'column_invisible': [('parent.is_contract', '=', False)]}"
+                />
+                <field
+                    name="date_end"
+                    attrs="{'column_invisible': [('parent.is_contract', '=', False)]}"
+                />
+            </xpath>
+            <xpath expr="//field[@name='order_line']/tree" position="attributes">
```

Comment:
I confirm what @jesusVMayor says
Maybe adding a tab in the sale configurator popup with contract configurator fields, only when sale_product_configurator is installed, otherwise it is ok as it is